### PR TITLE
JS Project To-Do List

### PR DESCRIPTION
**Pull request by Jhonatanssal**
This PR has an implementation of a To-Do list where a user can add projects and inside each project add specific tasks or steps needed to accomplish the project.

### The project was built using:
- JavaScript.
- Webpack.
- Eslint.
- Stylelint.
- CSS

Hope you like it.
Happy coding. 👍🏼

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,72 @@
...
 ## Built With
 
-- Major languages
-- Frameworks
-- Technologies used
+- JavaScript
+- Bootstrap
+- HTML5
+- Eslint
+- Stylelintrc
```
Comment (bellom) [inline README.md:13]: Kindly push these two files by removing them from `.gitignore`, we need to check the files.
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

File: README.md [hunk 2]
```diff
@@ -1,75 +1,72 @@
...
+npm
 
 ## Getting Started
 
-**This is an example of how you may give instructions on setting up your project locally.**
-**Modify this file to match your project, remove sections that don't apply. For example: delete the testing section if the currect project doesn't require testing.**
+To get a local copy up and running follow these simple example steps:
 
+- Go to https://github.com/Alexoid1/Library and clone the repo using: <br>
+`git clone https://github.com/Alexoid1/Library.git` in your terminal.
```
Comment (bellom) [inline README.md]: **Kindly correct the information you're having inside this README file.**
_1. you have incorrect repo link_
_2. you have an incorrect live link._

File: dist/index.html [hunk 3]
```diff
@@ -4,10 +4,34 @@
...
+    <div id='projectsContainer'>
+      
+
+    </div>
+    
+    <div id='modal'>
+      
+    </div>
+
+  </div>
```
Comment (bellom) [inline dist/index.html]: Kindly clear up your codebase by removing unnecessary spaces.

File: package.json [hunk 4]
```diff
@@ -17,6 +17,10 @@
   },
   "homepage": "https://github.com/jssarrazolaa/JS-ToDoApp#readme",
```
Comment (bellom) [inline package.json:20]: We need to add some scripts which are necessary to build your project;
Kindly look into your webpack set up, I should be the one to generate the `dist` folder by running `npm run build`, but it has been hardcoded by you.

example of a script to add;
```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },

```


image

https://microverse.pathwright.com/library/javascript-module-v10-copy/166786/path/step/97635165/

File: src/createProject.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
+/* eslint-disable import/no-cycle */
```
Comment (bellom) [inline src/createProject.js]: With the ESLint inline comment, we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

What could be the reason? If you are using VSCode, you can delete this line and hover over the error(configure eslint checks locally). And you will see your options. Click on the more info button that will take you to a GitHub repo where it is explained why you are getting this error. I find this feature very educational. Please, try to learn and implement the necessary rules so you don't have to keep this disabling line on top of each file. I believe you can do this! 💪😇

File: src/edit.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable import/no-cycle */
```
Comment (bellom) [inline src/edit.js]: With the ESLint inline comment, we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

File: src/editForm.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable import/no-cycle */
```
Comment (bellom) [inline src/editForm.js]: With the ESLint inline comment, we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

File: src/createProject.js [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+      form.appendChild(contI2);
+      form.appendChild(contI3);
+      form.appendChild(contI4);
+      form.appendChild(contI5);
+      form.appendChild(input6);
+      contF.appendChild(form);
+      modall.appendChild(contF);
+      input6.addEventListener('click', () => {
+        if (input1.value === '' && input2.value === '' && input3.value === '') {
+          alert("Title, description an date can't be empty");
```
Comment (bellom) [inline src/createProject.js]: Kindly remove `alert`.
Do not use Window.alert() or Window.confirm(). It’s ugly, impossible to style, stop code execution, and display differently on different browsers. Use custom modal instead;
image

https://microverse.pathwright.com/library/javascript-module-v10-copy/166786/path/step/97635165/

File: src/editForm.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  form.appendChild(contI2);
+  form.appendChild(contI3);
+  form.appendChild(contI4);
+  form.appendChild(contI5);
+  form.appendChild(input6);
+  contF.appendChild(form);
+  modall.appendChild(contF);
+  input6.addEventListener('click', () => {
+    if (input1.value === '' && input2.value === '' && input3.value === '') {
+      alert("Title, description an date can't be empty");
```
Comment (bellom) [inline src/editForm.js]: Kindly remove `alert`.

File: src/index.js [hunk 10]
```diff
@@ -1 +1,22 @@
...
+
+const botonProject = document.getElementById('botonProject');
+
+function initTodo() {
+  botonProject.addEventListener('click', () => {
+    const nameProject = document.getElementById('projectName').value;
+    const project = new Project(nameProject);
+
+    if (nameProject === '') {
+      alert('Project has to have a name!');
```
Comment (bellom) [inline src/index.js]: Kindly remove `alert`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/index.js (hunk 10 above)
Comment (Jhonatanssal) [reply src/index.js]: Thank you very much for the review, have a nice day. 👍🏼

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
#### Hi @jssarrazolaa and @Alexoid1 :clap: :clap:
#### You have done excellent work :muscle: by implementing the required changes. However, the app was unable to run using the `npm run start` command. Additionally, since `webpack-dev-server` is part of the npm script you should add the dependency.

#### _Reviewed by Temitayo Adeleke_

**Comment by bellom** [comment]
Hi Guys, let's look into the webpack file.
I'm here if you need help....

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**